crates/sui-graphql-rpc/src/types/base64.rs

```diff
@@ -1,18 +1,43 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+use std::str::FromStr;
+
 use async_graphql::*;
+use fastcrypto::encoding::Encoding;
 
-struct Base64(String);
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub(crate) struct Base64(String);
 
 /// TODO: implement Base64 scalar type
 #[Scalar]
 impl ScalarType for Base64 {
-    fn parse(_value: Value) -> InputValueResult<Self> {
-        unimplemented!()
+    fn parse(value: Value) -> InputValueResult<Self> {
+        match value {
+            Value::String(s) => {
+                fastcrypto::encoding::Base64::decode(&s)
+                    .map_err(|_| InputValueError::custom("Invalid Base64"))?;
+                Ok(Base64(s))
+            }
+            _ => Err(InputValueError::custom("Invalid Base64")),
```

Comment:
nit: Let's distinguish the error wording for the case where the value is a string with invalid Base64 in it, vs a non-String scalar value.